server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -126,28 +129,20 @@ func (p *FetchServer) FetchBlob(ctx context.Context, req *rapb.FetchBlobRequest)
 		}
 	}
 	httpClient := timeoutHTTPClient(ctx, req.GetTimeout())
+
+	// Keep track of the last fetch error so that if we fail to fetch, we at
+	// least have something we can return to the client.
+	var lastFetchErr error
+
 	for _, uri := range req.GetUris() {
 		_, err := url.Parse(uri)
 		if err != nil {
-			return nil, status.InvalidArgumentErrorf("Unparsable URI: %q", uri)
+			return nil, status.InvalidArgumentErrorf("unparsable URI: %q", uri)
 		}
-		rsp, err := httpClient.Get(uri)
+		blobDigest, err := mirrorToCache(ctx, cache, httpClient, uri, expectedSHA256)
 		if err != nil {
-			return nil, status.AbortedErrorf("Error fetching URI  %q: %s", uri, err.Error())
-		}
-		data, err := io.ReadAll(rsp.Body)
-		if err != nil {
-			log.Warningf("Error reading object bytes: %s", err.Error())
-			continue
-		}
-		reader := bytes.NewReader(data)
-		blobDigest, err := digest.Compute(reader)
-		if err != nil {
-			log.Warningf("Error computing object digest: %s", err.Error())
-			continue
-		}
-		if err := cache.Set(ctx, blobDigest, data); err != nil {
-			log.Warningf("Error inserting object into cache: %s", err.Error())
+			lastFetchErr = err
+			log.Warning(status.Message(err))
```

Comment:
Can you decorate this error message a bit so that it's clear where the error is coming from?
Something like log.Warningf("could not mirror URL to cache: %s", err)

In general I recommend ever having undecorated logs like log.Warning(err).

Reply:
Done